lock/src/test/java/org/infinispan/lock/impl/lock/RequestExpirationSchedulerTest.java

```diff
@@ -0,0 +1,123 @@
+package org.infinispan.lock.impl.lock;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertFalse;
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.lock.BaseClusteredLockTest;
+import org.infinispan.lock.api.ClusteredLockManager;
+import org.infinispan.lock.impl.manager.EmbeddedClusteredLockManager;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "clusteredLock.RequestExpirationSchedulerTest")
+public class RequestExpirationSchedulerTest extends BaseClusteredLockTest {
+
+   private ClusteredLockManager cm;
+   private RequestExpirationScheduler expirationScheduler;
+
+   @Override
+   protected int clusterSize() {
+      return 1;
+   }
+
+   @BeforeClass
+   public void createLockManager() throws Throwable {
+      cm = clusteredLockManager(0);
+   }
+
+   @BeforeMethod
+   public void createRequestExpirationScheduler() {
+      EmbeddedClusteredLockManager eclm = (EmbeddedClusteredLockManager) cm;
+      expirationScheduler = new RequestExpirationScheduler(eclm.getScheduledExecutorService());
+   }
+
+   @Test
+   public void testScheduleForCompletion() throws Exception {
+      CompletableFuture<Boolean> request = new CompletableFuture();
+
+      expirationScheduler.scheduleForCompletion("123", request, 10, TimeUnit.MILLISECONDS);
```

Comment:
I wouldn't trust these tests with such low timeouts. I can guarantee some of these will fail at some point in the future. I would instead use Mockito to mock out the `ScheduledExecutorService` and verify invocations on it. This way the test should be 100% deterministic and should be pretty quick not having to wait for executions at all.

Reply:
I considered using mockito and verify at first but I reconsidered to test cancel status etc.... but ok  to use mockito if you think it will probably fail in the furure

Reply:
We still test cancellation :) It is just that you invoke it (like the scheduled executor would have) But you get to control when it is invoked. Our code should assume that scheduled executor works properly and this way we don't have to have weird timeouts or wait for such things to happen.